Automattic/jetpack
# Social: Refactor the TemplatePicker component

This change refactors our current `TemplatePicker` component, so that the inner logic can be used separately, decoupled from the modal. This is needed so we can use the picker in the SIG modal for #31665 .

## Proposed changes:

* Created new component for the picker itself. It has a `onTemplateSelected` prop, which can be used by the parent component to commit the changes, or do something with them.
* Used this new component inside our old one, which was renamed to `TemplatePickerModal`.

### Other information:


## Jetpack product discussion


1204930070946424-as-1204986292229845
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* Open a new post with the Advanced plan.
  * Check that the template picker works as before:
    * It should open up correctly, and select the first - default - template.
    * Changing a template and saving should be reflected on the preview.
    * Re-opening the picker should show the selected template.
    * Publishing the post with SIG should use the correct image.
 * On the admin page, the template selector and changing default templates should work too.
